### PR TITLE
Add ekuznetsov139's patch for the 3.0 compilation issue.

### LINKED ISSUE #283: Build fails with compiler error on rocm 3.0
When compiling with hcc, via, for example,
```
cmake -DCMAKE_CXX_COMPILER=hcc ..
```
the compilation fails with some stack trace errors in '/opt/rocm/hcc/bin/clang-10'.

### PR DESCRIPTION
**Pull request by malcolmroberts**
resolves #283

### CODE REVIEW

**Comment by malcolmroberts** [comment]
Failures seem to be related to docker issues and not htis patch.

File: library/src/device/generator/generator.pass.hpp [hunk 1]
```diff
@@ -844,31 +844,19 @@ namespace StockhamGenerator
...
-                                    passStr += regRealIndex;
-                                    passStr += ") - (W.y * ";
-                                    passStr += regImagIndex;
-                                    passStr += ");\n\t\t";
-                                    passStr += "TI = (W.y * ";
-                                    passStr += regRealIndex;
-                                    passStr += ") + (W.x * ";
-                                    passStr += regImagIndex;
-                                    passStr += ");\n\t\t";
+                                    passStr += "TR = wx * rx - wy * ry;\n\t\t";
```
Comment (saadrahim) [inline library/src/device/generator/generator.pass.hpp]: You are just rearranging expressions so the compiler bug is avoided?

**Comment by malcolmroberts** [comment]
[figs.pdf](https://github.com/ROCmSoftwarePlatform/rocFFT/files/4010847/figs.pdf)
Looks like there's no effect on performance; so that's good!

File: library/src/device/generator/generator.pass.hpp (hunk 1 above)
Comment (malcolmroberts) [reply library/src/device/generator/generator.pass.hpp]: There's a local variable that serves as an explicit cast.

**Pull request merged**